packages/bitcore-node/src/models/transaction.ts

```diff
@@ -274,49 +282,179 @@ export class TransactionModel extends BaseTransaction<IBtcTransaction> {
     const rpc = new AsyncRPC(username, password, host, port);
     try {
       const result = await rpc.call('gettransactionreceipt', [txid]);
-      if (result.length > 0 && result[0].contractAddress) {
+      if (result.length > 0 && result[0].contractAddress && result[0].log.length === 0) {
         const contractAddress = result[0].contractAddress;
-        let name = '';
-        let symbol = '';
-        let totalSupply = 0;
         const decimals = await rpc.call('frc20decimals', [contractAddress]);
-        if (decimals !== 0) {
-          name = await rpc.call('frc20name', [contractAddress]);
-          symbol = await rpc.call('frc20symbol', [contractAddress]);
+        const name = await rpc.call('frc20name', [contractAddress]);
+        const symbol = await rpc.call('frc20symbol', [contractAddress]);
+        let totalSupply = '0';
+        try {
           totalSupply = await rpc.call('frc20totalsupply', [contractAddress]);
+        } catch (error) {
+          if ((error as any).message && (error as any).message !== 'Integer Division by zero.') {
+            throw error;
+          }
+        }
+        const checkIsERC20 = decimals !== 0 && name !== '' && symbol !== '' && totalSupply !== '0';
+        if (checkIsERC20) {
+          const totalSupplyDecimal = countDecimals(+totalSupply);
+          totalSupply = (
+            BigInt(new Decimal(totalSupply).mul(new Decimal(10 ** totalSupplyDecimal)).toString()) *
```

Comment:
10 ** totalSupplyDecimal could exceed 53 bytes also, please use something like `new Decimal(10).pow(totalSupplyDecimal)`

Reply:
Or create util function to convert this

Reply:
ok kub